vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ReferenceManagerTest.java

```diff
@@ -0,0 +1,339 @@
+                Sort.by(SortByField.BIBLIO_TITLE.getValue()).descending());
+        when(refRepo.count()).thenReturn(1L);
+        when(refRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<Reference>(refList));
+        List<IReference> requestedReferences = refManagerToTest.getReferences(1,
+                SortByField.BIBLIO_TITLE.getValue(), Sort.Direction.DESC.toString());
+        assertEquals(true, checkSortByTitleDesc(requestedReferences));
+        assertEquals(2, requestedReferences.size());
+        assertEquals(REF_ID2, requestedReferences.get(0).getId());
+    }
+
+    @Test
+    public void test_getReferences_sorted_negativePage() {
+        Pageable sortByRequestedField = PageRequest.of(0, 10,
+                Sort.by(SortByField.BIBLIO_TITLE.getValue()).descending());
+        when(refRepo.count()).thenReturn(1L);
+        when(refRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<Reference>(refList));
+        List<IReference> requestedReferences = refManagerToTest.getReferences(-2,
+                SortByField.BIBLIO_TITLE.getValue(), Sort.Direction.DESC.toString());
+        assertEquals(true, checkSortByTitleDesc(requestedReferences));
+        assertEquals(REF_ID2, requestedReferences.get(0).getId());
+    }
+
+    @Test
+    public void test_getReferences_sorted_pageGreaterThanTotalPages() {
+        ReflectionTestUtils.setField(refManagerToTest, "pageSize", 1);
+        Pageable sortByRequestedField = PageRequest.of(4, 1, Sort.by(SortByField.BIBLIO_TITLE.getValue()).descending());
+        when(refRepo.count()).thenReturn(5L);
+        when(refRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<Reference>(refList));
+        List<IReference> requestedReferences = refManagerToTest.getReferences(7,
+                SortByField.BIBLIO_TITLE.getValue(), Sort.Direction.DESC.toString());
+        assertEquals(true, checkSortByTitleDesc(requestedReferences));
+        assertEquals(REF_ID2, requestedReferences.get(0).getId());
+    }
+
+    @Test
+    public void test_getReferences_sorted_noResult() {
+        ReflectionTestUtils.setField(refManagerToTest, "pageSize", 1);
+        Pageable sortByRequestedField = PageRequest.of(4, 1,
+                Sort.by(SortByField.CREATION_DATE.getValue()).descending());
+        when(refRepo.count()).thenReturn(5L);
+        when(refRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<Reference>(new ArrayList<>()));
+        List<IReference> requestedReferences = refManagerToTest.getReferences(1,
+                SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
+        assertEquals(true, requestedReferences.isEmpty());
+    }
+
+    @Test
+    public void test_getTotalBiblioCount_success() {
+        when(refRepo.count()).thenReturn(5L);
+        assertEquals(5L, refManagerToTest.getTotalReferenceCount());
+    }
+
+    @Test
+    public void test_getTotalBiblioCount_whenZeroBiblios() {

```

Comment:
why `whenZeroBiblios`? shouldn't that be references?

Reply:
Yes corrected!